src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -350,53 +353,52 @@ public World[] executeSimple(Object[][] params) {
-					"teleport player to location(1,1,1, world of player)",
-					"teleport player to location(1,1,1, world(\"world\"))",
-					"teleport player to location({_x}, {_y}, {_z}, {_w}, {_yaw}, {_pitch})",
-					"# SETTING BLOCKS",
-					"set block at location(1,1,1, world \"world\") to stone",
-					"set block at location(1,1,1, world \"world\", 100, 0) to stone",
-					"set block at location(1,1,1, world of player) to stone",
-					"set block at location(1,1,1, world(\"world\")) to stone",
-					"set block at location({_x}, {_y}, {_z}, {_w}) to stone",
-					"# USING VARIABLES",
-					"set {_l1} to location(1,1,1)",
-					"set {_l2} to location(10,10,10)",
-					"set blocks within {_l1} and {_l2} to stone",
-					"if player is within {_l1} and {_l2}:",
-					"# OTHER",
-					"kill all entities in radius 50 around location(1,65,1, world \"world\")",
-					"delete all entities in radius 25 around location(50,50,50, world \"world_nether\")",
-					"ignite all entities in radius 25 around location(1,1,1, world of player)")
-			.since("2.2"));
+		Functions.register(DefaultFunction.builder("location", Location.class)
+			.description(
+				"Creates a location from a world and 3 coordinates, with an optional yaw and pitch.",
+				"If for whatever reason the world is not found, it will fallback to the server's main world.")
+			.examples(
+				"""
+					# TELEPORTING
+					teleport player to location(1,1,1, world "world")
+					teleport player to location(1,1,1, world "world", 100, 0)
+					teleport player to location(1,1,1, world "world", yaw of player, pitch of player)
+					teleport player to location(1,1,1, world of player)
+					teleport player to location(1,1,1, world("world"))
+					teleport player to location({_x}, {_y}, {_z}, {_w}, {_yaw}, {_pitch})
+					
+					# SETTING BLOCKS
+					set block at location(1,1,1, world "world") to stone
+					set block at location(1,1,1, world "world", 100, 0) to stone
+					set block at location(1,1,1, world of player) to stone
+					set block at location(1,1,1, world("world")) to stone
+					set block at location({_x}, {_y}, {_z}, {_w}) to stone
+					
+					# USING VARIABLES
+					set {_l1} to location(1,1,1)
+					set {_l2} to location(10,10,10)
+					set blocks within {_l1} and {_l2} to stone
+					if player is within {_l1} and {_l2}:
+					
+					# OTHER
+					kill all entities in radius 50 around location(1,65,1, world "world")
+					delete all entities in radius 25 around location(50,50,50, world "world_nether")
+					ignite all entities in radius 25 around location(1,1,1, world of player)
+					""")
+			.since("2.2")
+			.parameter("x", Number.class)
+			.parameter("y", Number.class)
+			.parameter("z", Number.class)
+			.optionalParameter("world", World.class)
+			.optionalParameter("yaw", Float.class)
+			.optionalParameter("pitch", Float.class)
+			.build(args -> {
+				World world = args.getOrDefault("world", Bukkit.getWorlds().get(0));
```

Comment:
This doesn't mirror the previous implementation's behavior where it would try to use the world event value as a default value if no world argument was passed in

Reply:
Do you think it's possible for functions to be aware of the context their in? Maybe store the calling event in the FuntionEvent object? That'll allow event values to be used for optional parameters and such, which I think is more intuitive.